src/Optimizer/Statistics/Histogram.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <base/types.h>
+
+namespace DB
+{
+
+struct Bucket;
+using Buckets = std::vector<Bucket>;
+
+struct Bucket
+{
+    /// Lower bound of the bucket.
+    Float64 lower;
+    /// Upper bound of the bucket.
+    Float64 upper;
+    /// How many rows in the bucket.
+    Float64 count;
+    /// Number of distinct value in the bucket.
+    Float64 ndv;
+};
+
+/// Histogram of a column, used to eliminate row count.
+/// Not used right now
+class Histogram
+{
+public:
+    Histogram(Buckets buckets_) : buckets(std::move(buckets_)) { }
+
+    Float64 eliminateEA(Float64 value);
```

Comment:
what `eliminate` means and where it is implemented?

Reply:
Histogram is not implemented right now, there is a comment in class header. I will delete it later.